### PR TITLE
Event documents transformer

### PR DESCRIPTION
**Pull request by agnesgaroux**
[Issue](https://github.com/wellcomecollection/content-api/issues/64) 

- added contributors to the eventDoc query: that makes its type the same as series on articles
- ~the above could actually be removed, since~ I ended up not including `contributors` in `display` or `query`. Reason is, it doesn't seem important enough to display on the result list, and I don't see people querying for specific hosts or facilitator. Happy to discuss 💬
- Any idea what we could use as "default event format"? it's sometimes undefined 
- not sure about the `times` in the `display`

### CODE REVIEW

**Comment by rcantin-w** [comment]
> the above could actually be removed, since I ended up not including contributors in display or query. Reason is, it doesn't seem important enough to display on the result list, and I don't see people querying for specific hosts or facilitator. Happy to discuss 💬

I thiiiink this was because of curators or specific artists, but tbh I'm not sure! Happy to leave out for now and add later if we want it back?

File: pipeline/src/transformers/eventDocument.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  isImageLink,
+  asText,
+  asTitle,
+  isNotUndefined,
+} from "../helpers/type-guards";
+import { linkedDocumentIdentifiers, formatSeriesForQuery } from "./utils";
+
+const onlineLocation = {
+  type: "EventLocation",
+  id: "ef04c8e3-26be-4fbc-9bef-f52589ebc56c",
```
Comment (rcantin-w) [inline pipeline/src/transformers/eventDocument.ts:25]: Where is this id from? (not Prismic?)

File: pipeline/src/types/transformed/eventDocument.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { Image } from ".";
+
+export type EventDocument = {
+  type: "Event";
+  id: string;
+  title: string;
+  image?: Image;
+  times: { startDateTime: Date | null; endDateTime: Date | null }[];
+  format: EventDocumentFormat | undefined;
```
Comment (rcantin-w) [inline pipeline/src/types/transformed/eventDocument.ts:9]: Could we do `format?: EventDocumentFormat;` to match the others?

File: pipeline/src/types/transformed/eventDocument.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { Image } from ".";
+
+export type EventDocument = {
+  type: "Event";
+  id: string;
+  title: string;
+  image?: Image;
+  times: { startDateTime: Date | null; endDateTime: Date | null }[];
```
Comment (rcantin-w) [inline pipeline/src/types/transformed/eventDocument.ts:8]: Can we do the same here or does it have to be `null`?

File: pipeline/src/transformers/eventDocument.ts (hunk 1 above)
Comment (agnesgaroux) [reply pipeline/src/transformers/eventDocument.ts]: As discussed, isOnline is a boolean in Prismic. We want to treat it as a location so it needs to be in the same format as physical location

File: pipeline/src/types/transformed/eventDocument.ts (hunk 3 above)
Comment (agnesgaroux) [reply pipeline/src/types/transformed/eventDocument.ts]: Aligning on how the FE handles this, it is now optional to account for the possibility of it being undefined (instead of null)

File: common/data/defaultValues.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+// used for display purpose only, when the prismic data is undefined
+export const defaultArticleFormat = {
+  type: "ArticleFormat",
+  id: "W7TfJRAAAJ1D0eLK",
+  label: "Article",
+};
+
+export const defaultEventFormat = {
+  type: "EventFormat",
+  id: "dfc2b7f9-c362-47da-9644-b0f98212ccaa",
```
Comment (rcantin-w) [inline common/data/defaultValues.ts:10]: If we're keeping this I'd be tempted to add a comment to explain the difference in IDs

File: pipeline/src/graph-queries/eventDocuments.ts [hunk 5]
```diff
@@ -61,6 +61,16 @@ const query = `{
     series {
       series {
         title
+        contributors {
```
Comment (rcantin-w) [inline pipeline/src/graph-queries/eventDocuments.ts:64]: Is this what it related to? 
https://github.com/wellcomecollection/content-api/pull/75#issuecomment-1824118943

**Review by rcantin-w — APPROVED** [review]

File: pipeline/src/graph-queries/eventDocuments.ts (hunk 5 above)
Comment (agnesgaroux) [reply pipeline/src/graph-queries/eventDocuments.ts:64]: Ha! no I got the original commit comment wrong (fixed it now) because I confused contributors and series contributors
The EventDocument query gets:
- Contributors, which I left out of the transformed document because it doesn't seem very useful for either display or query
- Series, which now includes Series contributors, so it's aligned with Articles

Comment (rcantin-w) [reply pipeline/src/graph-queries/eventDocuments.ts:64]: So we're fetching Contributors but not using them afterwards at the moment?

File: common/data/defaultValues.ts (hunk 4 above)
Comment (agnesgaroux) [reply common/data/defaultValues.ts:12]: ✅

File: pipeline/src/graph-queries/eventDocuments.ts (hunk 5 above)
Comment (agnesgaroux) [reply pipeline/src/graph-queries/eventDocuments.ts:64]: Indeed. The idea is the query has pretty much everything we _might_ want to use, we can trim it down once the transformed type is a bit more stable/developed to remove anything we don't actually want in the index

Comment (rcantin-w) [reply pipeline/src/graph-queries/eventDocuments.ts:64]: Works for me :)
